src/get_information.cpp

```diff
@@ -218,6 +218,96 @@ InfoBlock getCondMutualInfo(int X, int Y, const vector<int>& ui_list,
   return res;
 }
 
+double getEntropy(Environment& environment, int Z, int X, int Y) {
+  TempAllocatorScope scope;
+  auto& cache = environment.cache.info_score;
+
+  double H{0};
+  bool found{false};
+  vector<int> ui_list; 
+  std::tie(H, found) = cache->getEntropy(X, Y, Z, ui_list);
+  if (found) return H;
+
+  bool any_na = environment.has_na[X] || environment.has_na[Y] ||
+                environment.has_na[Z];
+  // Mark rows containing NAs and count the number of complete samples
+  // 1: sample contains no NA, 0: sample contains NA
+  TempVector<int> sample_is_not_NA(environment.n_samples, 1);
+  // vector with the number of rows containing NAs seen at rank i
+  TempVector<int> na_count(environment.n_samples, 0);
+  int n_samples_non_na_z = environment.n_samples;
+  if (any_na)
+    n_samples_non_na_z = countNonNA(
+        X, Y, Z, ui_list, environment.data_numeric, sample_is_not_NA, na_count);
```

Comment:
See the point 1. of the overall review. If `ui_list` is intended to be empty, better use a temp empty vector `vector<int>` here to be explicit, same for all other `ui_list`s in this function.